perfkitbenchmarker/providers/openstack/os_virtual_machine.py

```diff
@@ -162,14 +171,75 @@ def WaitForBootCompletion(self):
         if self.hostname is None:
             self.hostname = resp[:-1]
 
-    def CreateScratchDisk(self, disk_spec):
-        disks_names = ('%s-data-%d-%d'
-                       % (self.name, len(self.scratch_disks), i)
-                       for i in range(disk_spec.num_striped_disks))
-        disks = [os_disk.OpenStackDisk(disk_spec, name, self.zone)
-                 for name in disks_names]
+    def OnStartup(self):
+        super(OpenStackVirtualMachine, self).OnStartup()
+        self.boot_device = self._GetBootDevice()
 
-        self._CreateScratchDiskFromDisks(disk_spec, disks)
+    def CreateScratchDisk(self, disk_spec):
+        """
+        Creates instances of OpenStackDisk child classes depending on disk type.
+        """
+        if disk_spec.disk_type == os_disk.ROOT:  # Ignore num_striped_disks
+            if self.root_disk_allocated:
+                raise errors.Error('Only a Root disk can be created per VM')
+            device_path = '/dev/%s' % self.boot_device['name']
```

Comment:
This could be `self.boot_device.GetDevicePath()` if you accept my comment above about not throwing an exception in `GetDevicePath()`.